jwql/tests/test_badpix_monitor.py

```diff
@@ -0,0 +1,216 @@
+    ratefiles, ratefiles2copy = bad_pixel_monitor.locate_rate_files(uncal_files)
+
+    rates = [os.path.basename(entry) for entry in ratefiles]
+    rates2copy = [os.path.basename(entry) for entry in ratefiles2copy]
+
+    expected = ['jw00300002001_02102_00001_mirimage_rateints.fits', 'jw00300002001_0210a_00001_mirimage_rateints.fits']
+    assert rates == expected
+    assert rates2copy == expected
+
+
+@pytest.mark.skipif(ON_JENKINS,
+                    reason='Requires access to central storage.')
+def test_locate_uncal_files():
+    """Test the filesystem search for uncal files
+    """
+    file1 = 'jw00300002001_02102_00001_mirimage_rate.fits'
+    file2 = 'jw00300010001_02102_00001_mirifushort_uncal.fits'
+    query_results = [{'filename': file1},
+                     {'filename': file2}]
+
+    found = bad_pixel_monitor.locate_uncal_files(query_results)
+
+    found_base = [os.path.basename(entry) for entry in found]
+    assert found_base[0] == file1.replace('rate', 'uncal')
+    assert found_base[1] == file2
+
+
+def test_filter_query_results():
+    """Test MAST query filtering to extract most common filter/pupil and
+    acceptable readout patterns
+    """
+    badpix = bad_pixel_monitor.BadPixels()
+    badpix.instrument = 'nircam'
+
+    dict1 = {'filter': 'F070W', 'pupil': 'CLEAR', 'readpatt': 'RAPID'}
+    dict2 = {'filter': 'F090W', 'pupil': 'CLEAR', 'readpatt': 'BRIGHT1'}
+    dict3 = {'filter': 'F070W', 'pupil': 'CLEAR', 'readpatt': 'BRIGHT1'}
+    query_results = [dict1, dict2, dict1, dict1, dict3, dict1]
+
+    filtered = badpix.filter_query_results(query_results, 'flat')
+
+    assert filtered == [dict1, dict1, dict1, dict1]
+
+    # NIRSpec
+    badpix.instrument = 'nirspec'
+
+    dict1 = {'filter': 'F070W', 'grating': 'CLEAR', 'readpatt': 'NRSRAPID'}
+    dict2 = {'filter': 'F070W', 'grating': 'CLEAR', 'readpatt': 'NRSIRS2'}
+    dict3 = {'filter': 'F070W', 'grating': 'CLEAR', 'readpatt': 'NRSRAPID'}
+    query_results = [dict1, dict2, dict1, dict1, dict3, dict1]
+
+    filtered = badpix.filter_query_results(query_results, 'flat')
+
+    assert filtered == [dict1, dict1, dict1, dict3, dict1]
+
+
+nrc_list = ['NRCA1_FULL', 'NRCB1_FULL', 'NRCA2_FULL', 'NRCB2_FULL', 'NRCA3_FULL', 'NRCB3_FULL',
+            'NRCA4_FULL', 'NRCB4_FULL', 'NRCA5_FULL', 'NRCB5_FULL']
+nis_list = ['NIS_CEN']
+nrs_list = ['NRS1_FULL', 'NRS2_FULL']
+miri_list = [('MIRIMAGE', 'MIRIM_FULL'), ('MIRIFULONG', 'MIRIM_FULL'), ('MIRIFUSHORT', 'MIRIM_FULL')]
+fgs_list = ['FGS1_FULL', 'FGS2_FULL']
+
+
+@pytest.mark.parametrize("instrument,expected_list", [("nircam", nrc_list), ("niriss", nis_list),
+                                                      ("nirspec", nrs_list), ("miri", miri_list), ("fgs", fgs_list)])
+def test_get_possible_apertures(instrument, expected_list):
+    """Make sure the correct apertures are returned for the given instrument
```

Comment:
Maybe a parameters section could be helpful here?

Reply:
I'm not sure what you mean here? I figured it was best to keep the creation of e.g. `nrc_list` close to `test_possible_apertures` since that is the only test that uses it.

Reply:
Okay, yes that makes sense!